### PR TITLE
Update intellij-idea-next-*-eap (2016.3 EAP) to Public Preview (build…

### PR DESCRIPTION
**Pull request by metyl**
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/intellij-idea-next-ce-eap.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 cask 'intellij-idea-next-ce-eap' do
-  version '2016.3,163.6110.12'
-  sha256 'ae35cf4d9cc057bfdf447b3cb771e2e9b8ab0ed76e3434e43ffd6771809fa543'
+  version '2016.3,163.6512.17'
+  sha256 '3200510012a08ff0813100012e09a649b7695fa4f0d16c6c30bd86da76e4d66d'
 
-  url "https://download.jetbrains.com/idea/ideaIC-#{version.after_comma}.dmg"
+  url 'https://download.jetbrains.com/idea/ideaIC-2016.3-PublicPreview.dmg'
```
Comment (vitorgalvao) [inline Casks/intellij-idea-next-ce-eap.rb]: Use `#{version.before_comma}` instead.

File: Casks/intellij-idea-next-eap.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 cask 'intellij-idea-next-eap' do
-  version '2016.3,163.6110.12'
-  sha256 'c7afe1ee51260824a814c5950dd61f10e7119475bd2a233d89a419b2ff47eb84'
+  version '2016.3,163.6512.17'
+  sha256 '58404b598830fdb6b2979572327fafa17ea6160dcd99f88d576536172987ef3b'
 
-  url "https://download.jetbrains.com/idea/ideaIU-#{version.after_comma}.dmg"
+  url 'https://download.jetbrains.com/idea/ideaIU-2016.3-PublicPreview.dmg'
```
Comment (vitorgalvao) [inline Casks/intellij-idea-next-eap.rb]: Same here.

Comment (metyl) [inline Casks/intellij-idea-next-eap.rb]: Thx, applied in e883285b6c6be6b50beae04c657ab91ab1062075.

**Comment by mtraynham** [comment]
The next-eap breaks with this change.  The download URL is correct, but the app reference is wrong.

https://github.com/metyl/homebrew-versions/blob/e883285b6c6be6b50beae04c657ab91ab1062075/Casks/intellij-idea-next-eap.rb#L11

Per that line, it's expecting `IntelliJ IDEA 2016.3-PublicPreview EAP.app`, but if you download the dmg, the app is still named `IntelliJ IDEA 2016.3 EAP.app`.

**Comment by metyl** [comment]
@mtraynham thx for feedback, the initial version was ok but the braking change was introduced by e883285b6c6be6b50beae04c657ab91ab1062075. 
I've made PR #2797 to fix it.

**Comment by mtraynham** [comment]
Thanks @metyl !

**Pull request merged**